kratos/containers/pointer_vector_set.h

```diff
@@ -1072,44 +1083,43 @@ class PointerVectorSet final
                 mData.push_back(TPointerType(&GetReference(it)));
             }
         } else {
-            // first find the largest range
             const auto lower_bound_first = std::lower_bound(mData.begin(), mData.end(), KeyOf(GetReference(first)), CompareKey());
```

Comment:
why don't we take directly the begin?

Reply:
i mean somethign like 

if(KeyOf(GetReference(first)) > KeyOf(mData().end()-1))

which has O(1) cost instead of O(logN)

Reply:
I missed this. I'll change.

Reply:
still missing this one i believe, other than that ok on my side